cli/azd/internal/grpcserver/compose_service.go

```diff
@@ -0,0 +1,234 @@
+}
+
+// ListResources lists all resources in the project configuration.
+func (c *composeService) ListResources(
+	ctx context.Context,
+	req *azdext.EmptyRequest,
+) (*azdext.ListResourcesResponse, error) {
+	azdContext, err := c.lazyAzdContext.GetValue()
+	if err != nil {
+		return nil, err
+	}
+
+	projectConfig, err := project.Load(ctx, azdContext.ProjectPath())
+	if err != nil {
+		return nil, err
+	}
+
+	existingResources := projectConfig.Resources
+	composedResources := make([]*azdext.ComposedResource, 0, len(existingResources))
+
+	for _, resource := range existingResources {
+		resourceConfigBytes, err := json.Marshal(resource.Props)
+		if err != nil {
+			return nil, fmt.Errorf("marshaling resource config: %w", err)
+		}
+		composedResource := &azdext.ComposedResource{
+			Name:   resource.Name,
+			Type:   string(resource.Type),
+			Config: resourceConfigBytes,
+			Uses:   resource.Uses,
+		}
+		composedResources = append(composedResources, composedResource)
+	}
+
+	return &azdext.ListResourcesResponse{
+		Resources: composedResources,
+	}, nil
+}
+
+// createResourceProps unmarshals the resource configuration bytes into the appropriate struct based on the resource type.
+func createResourceProps(resourceType string, config []byte) (any, error) {
+	switch project.ResourceType(resourceType) {
+	case project.ResourceTypeHostContainerApp:
+		props := project.ContainerAppProps{}
+		if len(config) == 0 {
+			return props, nil
+		}
+		if err := json.Unmarshal(config, &props); err != nil {
+			return nil, err
+		}
+		return props, nil
+	case project.ResourceTypeDbCosmos:
+		props := project.CosmosDBProps{}
+		if len(config) == 0 {
+			return props, nil
+		}
+		if err := json.Unmarshal(config, &props); err != nil {
+			return nil, err
+		}
+		return props, nil
+	case project.ResourceTypeStorage:
+		props := project.StorageProps{}
+		if len(config) == 0 {
+			return props, nil
+		}
+		if err := json.Unmarshal(config, &props); err != nil {
+			return nil, err
+		}
+		return props, nil
+	case project.ResourceTypeAiProject:
+		props := project.AiFoundryModelProps{}
+		if len(config) == 0 {
+			return props, nil
+		}
+		if err := json.Unmarshal(config, &props); err != nil {
+			return nil, err
+		}
+		return props, nil
+	case project.ResourceTypeDbMongo:
+		props := project.CosmosDBProps{}
+		if len(config) == 0 {
+			return props, nil
+		}
+		if err := json.Unmarshal(config, &props); err != nil {
+			return nil, err
+		}
+		return props, nil
+	case project.ResourceTypeMessagingEventHubs:
+		props := project.EventHubsProps{}
+		if len(config) == 0 {
+			return props, nil
+		}
+		if err := json.Unmarshal(config, &props); err != nil {
+			return nil, err
+		}
+		return props, nil
+	case project.ResourceTypeMessagingServiceBus:
+		props := project.ServiceBusProps{}
+		if len(config) == 0 {
+			return props, nil
+		}
+		if err := json.Unmarshal(config, &props); err != nil {
+			return nil, err
+		}
+		return props, nil
+	default:
```

Comment:
Should this always match what's in `UnmarshalYAML` in `cli/azd/pkg/project/resources.go`? If so could we document this somehow?

Reply:
Yes, in the short term we'll need to keep this in sync.  I'll add a note in here.